src/depositions/Deposition.ts

```diff
@@ -78,4 +216,151 @@ export class Deposition {
+    const deposition = new Deposition(this.zenodo, await response.json());
+    this.zenodo.logger?.info(`Published deposition ${this.value.id}`);
+    return deposition;
+  }
+
+  /**
+   * Creates a new version of a published deposition.
+   * This is useful for updating the deposition with new files or metadata.
+   * The new version will have a new ID and will not overwrite the existing deposition.
+   * @returns Deposition - a new deposition object that is a new version of the current deposition
+   */
+  async newVersion(): Promise<Deposition> {
+    const response = await fetchZenodo(this.zenodo, {
+      route: `deposit/depositions/${this.value.id}/actions/newversion`,
+      method: 'POST',
+      expectedStatus: 201,
+    });
+    const deposition = new Deposition(this.zenodo, await response.json());
+    if (
+      deposition.value.metadata &&
+      !deposition.value.metadata?.publication_date
+    ) {
+      deposition.value.metadata.publication_date = new Date()
+        .toISOString()
+        .split('T')[0];
+      await deposition.update(deposition.value.metadata);
+    }
+    this.zenodo.logger?.info(
+      `Created new version for deposition ${this.value.id}`,
+    );
+    return deposition;
+  }
+
+  /**
+   * Upload a file to the deposition
+   * @param file - the file to upload to the deposition
+   * @returns ZenodoFile - the created file object
+   */
+  async uploadFile(file: File): Promise<ZenodoFile> {
+    const formData = new FormData();
+    formData.append('file', file);
+    const response = await fetchZenodo(this.zenodo, {
+      route: `deposit/depositions/${this.value.id}/files`,
+      method: 'POST',
+      body: formData,
+      expectedStatus: 201,
+    });
+    const zenodoFile = new ZenodoFile(this.zenodo, await response.json());
+    this.zenodo.logger?.info(
+      `Created file ${zenodoFile.value.id} for deposition ${this.value.id}`,
+    );
+    this.value.files?.push(zenodoFile.value);
+    return zenodoFile;
+  }
+
+  /**
+   * Submits the deposition for review by the community.
+   * @returns Deposition - the updated deposition object after submission
+   */
+  async submitForReview(): Promise<Deposition> {
+    const response = await fetchZenodo(this.zenodo, {
+      route: `records/${this.value.id}/actions/submit-review`,
+      method: 'POST',
+      expectedStatus: 202,
+    });
+    const deposition = new Deposition(this.zenodo, await response.json());
+    this.zenodo.logger?.info(
+      `Submitted deposition ${this.value.id} for review`,
+    );
+    return deposition;
+  }
+
+  /**
+   * Submits the deposition for review via browser.
+   * CORS request won't work from the other endpoint when using a browser.
+   * If a URL is provided, it will use that URL to submit the request.
+   * Otherwise, it will find the request for the current deposition and submit it.
+   * @param url - optional URL to submit the request
+   */
+  async submitForReviewBrowser(url?: string): Promise<void> {
```

Comment:
Rename to submitForReview and remove the current function of submitForReview. There should be a github link to this submitForReview previous code just in case they improve the API